### PR TITLE
Feature/assault validation

### PR DESCRIPTION
**Pull request by 7Z0nE**
implemented missing-feature #3
A faction can reach a planet, if it's solar system is connected to at least one solar system owned uniquely by that faction.
Also added tests for all involved features.
Since this is my very first (small) contribution, *feedback is highly appreciated*.

### CODE REVIEW

File: src/main/java/com/faforever/gw/model/ValidationHelper.java [hunk 1]
```diff
@@ -55,7 +55,7 @@ public void validateAssaultOnPlanet(GwCharacter character, Planet planet) {
             throw gwErrorService.getBpmnErrorOf(GwErrorType.PLANET_PROTECTED_FROM_ASSAULT);
         }
 
-        if (false) {// TODO: Implement check whether character's faction can reach planet
+        if (!planet.getSolarSystem().isReachable(character.getFaction())) {
```
Comment (Brutus5000) [inline src/main/java/com/faforever/gw/model/ValidationHelper.java:58]: This is not the only condition. The planet is also reachable if one of the other planets in the same solar systems is owned by the character's faction.

File: src/main/java/com/faforever/gw/model/SolarSystem.java [hunk 2]
```diff
@@ -89,4 +89,17 @@ public Faction uniqueOwner() {
 
         return uniqueFaction;
     }
+
+    @JsonIgnore
+    public boolean isReachable(Faction faction) {
+
+        for (SolarSystem system : getConnectedSystems()) {
```
Comment (Brutus5000) [inline src/main/java/com/faforever/gw/model/SolarSystem.java]: please add a note above that this approach only works with bidirectional links (just so we don't forget it later)

**Review by Brutus5000 — CHANGES_REQUESTED** [review]

**Comment by Brutus5000** [comment]
@7Z0nE Are you intending to implement the missing check (see my comment) or shall I merge your PR as is and leave the issue open?

**Comment by 7Z0nE** [comment]
I think I already did. See commit https://github.com/FAForever/gw-backend/commit/ecaac87541e544aa80bf009c10360321d9761c3b

**Pull request merged**